### PR TITLE
Omar manassra python 2 task

### PR DESCRIPTION
**Pull request by Omarmanassrah**
In update function in task_manager how i let the index if the list match the index i want to update

### CODE REVIEW

File: OmarManassra/__pycache__/task.cpython-312.pyc
Comment (Mohammed-samaheen) [inline OmarManassra/__pycache__/task.cpython-312.pyc:1]: We shouldn't include the Python cached files (Cython files). Please add the pycache folder to the .gitignore

File: OmarManassra/task_manager.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    description=input("enter the description")
+    due_date = input("Enter due date (YYYY-MM-DD): ")
+    validate_date(due_date)
+    status = input("Enter status ('InProgress' or 'Completed'): ")
+    validate_status(status)
+    
+    try:
+        """ Check if its Urgant """
+        isUrgant=input("enter if urgant or not (yes or no or Invalid)")
+        if isUrgant in ["yes","no"]:
```
Comment (Mohammed-samaheen) [inline OmarManassra/task_manager.py:18]: "is_urgent (bool): any shape of yes and no is considered as a valid input"

File: OmarManassra/task_manager.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+from  utils import validate_date,get_valid_input,validate_status,validate_priority
+from task import Task,UrgentTask
+# List contains all the tasks 
+tasks_list=[]
```
Comment (Mohammed-samaheen) [inline OmarManassra/task_manager.py:4]: why did you use a list to store your value, why not use a dictionary with the key task if and value the task itself?

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

**Comment by Omarmanassrah** [comment]
Done i will push the changes please see the changes and give a feedback
Thank you

File: OmarManassra/__pycache__/task.cpython-312.pyc
Comment (Mohammed-samaheen) [reply OmarManassra/__pycache__/task.cpython-312.pyc:1]: same issue

File: OmarManassra/Read.md
Comment (Mohammed-samaheen) [inline OmarManassra/Read.md:1]: Let's just stick with README.md as the file name

File: OmarManassra/.gitignore
Comment (Mohammed-samaheen) [inline OmarManassra/.gitignore:1]: git will not see this file, you should add it to the main directory

**Comment by Mohammed-samaheen** [comment]
resolve the conflict

File: OmarManassra/__pycache__/task.cpython-312.pyc
Comment (Mohammed-samaheen) [reply OmarManassra/__pycache__/task.cpython-312.pyc:1]: I added [.gitignore](https://github.com/AHL-Logics-backend-tranining-2024/python-task-2/blob/main/.gitignore) so please update your code, and remove the unwanted files.

File: OmarManassra/task.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    def display(self):
+        print (f"the title is{self.title} description {self.description}the due date {self.due_date},the status is {self.status}")
+
+# Adding Proiority 
+
+
+
+class UrgentTask(Task):
+    def __init__(self,priority,title,description,due_date,status="In progress"):
```
Comment (AMROAMRO404) [inline OmarManassra/task.py:27]: ``
```suggestion
    def __init__(self,priority,title,description,due_date,status="InProgress"):
```

**Comment by AMROAMRO404** [comment]
can you please use prettier for formatting the code ?

**Comment by AMROAMRO404** [comment]
can you resolve the conflict in .gitignore ?

File: OmarManassra/task_manager.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            title=input("Enter the title updated ")
+            description=input("enter the description update")
+            due_date = input("Enter due date (YYYY-MM-DD) update: ")
+            validate_date(due_date)
+            status = input("Enter status ('InProgress' or 'Completed'): update ")
+            validate_status(status)
+            task.title = title
+            task.description = description
+            task.due_date = due_date
+            task.status = status
```
Comment (AMROAMRO404) [inline OmarManassra/task_manager.py:68]: I see there are multiple lines duplicated, i think you can refactor it to mitigate this duplication.

File: OmarManassra/task_manager.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        Choice=input("Enter the choice ")
+        match Choice:
+            case "1":
+                 Add_task()
+            case "2":
+                 View_Tasks()
+            case "3":
+                 Update_Task()
+            case "4":
+                 Delete_Task()
```
Comment (AMROAMRO404) [inline OmarManassra/task_manager.py:93]: can you use lower_case for naming the methods ?

File: OmarManassra/task_manager.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    """ Call the Exception Handling Functions in the other module at every step it should be called"""
+    title=input("Enter the title")
+    description=input("enter the description")
+    due_date = input("Enter due date (YYYY-MM-DD): ")
+    validate_date(due_date)
+    status = input("Enter status ('InProgress' or 'Completed'): ")
+    validate_status(status)
+    try:
+        """ Check if its Urgant """
+        isUrgant=input("enter if urgant or not (True or False)")
```
Comment (AMROAMRO404) [inline OmarManassra/task_manager.py:15]: please use snack case for naming the variables.

File: OmarManassra/utils.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# The Date Exception Handling 
+def validate_date(date_str):
+    if date_str=="20-8-2024":
+        print ("the Date is right")
+    else:
+        raise ValueError("the value Date is not right ")
```
Comment (AMROAMRO404) [inline OmarManassra/utils.py:6]: we need to have a general validation not a specific date.

File: OmarManassra/__pycache__/task.cpython-312.pyc
Comment (AMROAMRO404) [inline OmarManassra/__pycache__/task.cpython-312.pyc:1]: can you remove this file.

File: OmarManassra/__pycache__/utils.cpython-312.pyc
Comment (AMROAMRO404) [inline OmarManassra/__pycache__/utils.cpython-312.pyc:1]: can you remove this file.

**Comment by Omarmanassrah** [comment]
@AMROAMRO404 
I made all the changes that is required if you can see it and give feedback if there is anything else
In .gitignore I accept both changes

File: .gitignore
Comment (Mohammed-samaheen) [inline .gitignore:1]: No need to add this file

Comment (Omarmanassrah) [reply .gitignore:1]: ??!

File: OmarManassra/task_manager.py
Comment (Momen-G-Ar) [inline OmarManassra/task_manager.py:1]: I think you have to install a spell checker, (recommended VSCode extension ID): `streetsidesoftware.code-spell-checker`

File: OmarManassra/utils.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+# The input Exception Handling
+def get_valid_input(prompt):
+    in_val=input(prompt)
+    if in_val:
+        print ("the value is Valid")
+    else:
+        raise ValueError("the Input Value is incorrect")
+# The Status Exception Handling 
+def validate_status(status):
+    if status=="Complate" or status=="InProgress":
```
Comment (Momen-G-Ar) [inline OmarManassra/utils.py:18]: `Completed` not `Complate`, the user will confused when he writes Completed and got an Error

File: OmarManassra/task_manager.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    try:
+        """ Check if its Urgant """
+        is_urgant=input("enter if urgant or not (True or False)")
+        if is_urgant:
+            priority = input("Enter priority ('High', 'Medium', 'Low'): ")
+            validate_priority(priority)
+            addUrgent= UrgentTask(priority,title,description,due_date,status)
+            tasks_dict[addUrgent.task_id]=addUrgent
+        else:
+            addtask=Task(title,description,due_date,status)
```
Comment (Momen-G-Ar) [inline OmarManassra/task_manager.py:22]: Please use snack case, `add_task`

File: OmarManassra/task_manager.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+def view_tasks():
+    """ Display the tasks"""
+    if tasks_dict:
+         for task in tasks_dict.values():
+            task.display()
+    else:
+        print ("No tasks added")
+
+
+def update_task():
```
Comment (Momen-G-Ar) [inline OmarManassra/task_manager.py:38]: What happens if the `task_id `is not in the `tasks_dict`? 
I think that the task manager should tell the user that the `id` does not exist before entering all this information

**Review by Momen-G-Ar — COMMENTED** [review]
There is a conflict in `.gitignore` file, please resolve it

**Comment by Omarmanassrah** [comment]
@Momen-G-Ar 
Momen I do not know the conflict is were can you give me more information in .gitignore which line is it the .gitignore in home directory

**Comment by Momen-G-Ar** [comment]
> @Momen-G-Ar Momen I do not know the conflict is were can you give me more information in .gitignore which line is it the .gitignore in home directory

image

**Comment by Omarmanassrah** [comment]
@Momen-G-Ar can you see if I fix it

**Review by Mohammed-samaheen — APPROVED** [review]

**Pull request merged**